Mathlib/Data/List/Sort.lean

```diff
@@ -70,6 +70,23 @@ theorem Sorted.tail {r : α → α → Prop} {l : List α} (h : Sorted r l) : So
 theorem rel_of_sorted_cons {a : α} {l : List α} : Sorted r (a :: l) → ∀ b ∈ l, r a b :=
   rel_of_pairwise_cons
 
+theorem Sorted.cons {r : α → α → Prop} [IsTrans α r] {l : List α} {a b : α}
+    (h : r b a) (ha : Sorted r (a :: l)) : Sorted r (b :: a :: l) := by
+  refine List.pairwise_cons.2 ⟨?_, ha⟩
+  rintro c (_ | _)
+  · exact h
+  · apply trans h
+    apply rel_of_sorted_cons ha
+    assumption
```

Comment:
```suggestion
nonrec theorem Sorted.cons {r : α → α → Prop} [IsTrans α r] {l : List α} {a b : α}
    (h : r b a) (ha : Sorted r (a :: l)) : Sorted r (b :: a :: l) :=
  ha.cons <| forall_mem_cons.2 ⟨h, fun _ hx => _root_.trans h <| rel_of_sorted_cons ha _ hx⟩
```

Reply:
Out of curiosity, what does the `nonrec` keyword do?

Reply:
It makes `.cons`  resolve to `Pairwise.cons` instead of the recursive `Sorted.cons`